libjulia/optimiser/CommonSubexpressionEliminator.cpp

```diff
@@ -33,9 +33,26 @@ using namespace dev::julia;
 
 void CommonSubexpressionEliminator::visit(Expression& _e)
 {
-	// Single exception for substitution: We do not substitute one variable for another.
-	if (_e.type() != typeid(Identifier))
-		// TODO this search rather inefficient.
+	DataFlowAnalyzer::visit(_e);
```

Comment:
The visit order is reversed now, because this will first substitute sub-expressions, which allows us to perform matching on outer expressions first.

OK, actually, let me turn this into a comment :)